Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp

```diff
@@ -155,9 +155,9 @@ void HeapSnapshotBuilder::analyzeIndexEdge(JSCell* from, JSCell* to, uint32_t in
     m_edges.append(HeapSnapshotEdge(from, to, index));
 }
 
-void HeapSnapshotBuilder::setOpaqueRootReachabilityReasonForCell(JSCell* cell, const char* reason)
+void HeapSnapshotBuilder::setOpaqueRootReachabilityReasonForCell(JSCell* cell, ASCIILiteral reason)
 {
-    if (!reason || !*reason || m_snapshotType != SnapshotType::GCDebuggingSnapshot)
+    if (reason.isNull() || reason.isEmpty() || m_snapshotType != SnapshotType::GCDebuggingSnapshot)
```

Comment:
Presumably `isEmpty` returns true for the null case too and we don’t have to also call `isNull`?